### ISSUE TITLE
The "run tests" button broke sometime after version 2.8.13 when test runner has a LOT of tests in it

### ISSUE THREAD

**Issue by rhymu8354** [matepek/vscode-catch2-test-adapter#179]
**Checklist**

- [X] The issue is about this extension and NOT about a fork.
- [X] Check the [known issues](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/documents/support.md#known-issues) list.
- [X] The latest version of the extension was used.

**Describe the bug**

The "run tests" button for a suite of tests no longer does anything when pressed.

Looking at the log, I think it has to do with some change made in how the test runner is launched.  I have a *LOT* of tests, so it looks like now with version 2.9.1 the command line is "too long".

**To Reproduce**

1. Set up a test runner with multiple test fixtures each having a *LOT* of tests.
2. Try to use the top-level "run tests" button in Test Explorer.
3. Nothing happens (there is a ENAMETOOLONG error in the log).

**Screenshots** (optional)

The following Twitch clip illustrates the problem:
* https://clips.twitch.tv/DrabNimbleTortoiseCeilingCat

The following Twitch clip illustrates how I fixed it by rolling back to 2.8.13:
* https://clips.twitch.tv/IcyConsiderateOctopusChefFrank

**Desktop**

- Extension Version: 2.9.1
- VS Code Version: 1.43.0
- Catch2 / Google Test / DOCTest Version:  googletest commit a18ac392
- OS Type and Version:  Windows 10.0.177633

**Log** (optional but recommended)

[catch2.log](https://github.com/matepek/vscode-catch2-test-adapter/files/4518388/catch2.log)

**Comment by matepek** [comment]
Thanks for the bug report.
Sorry for the inconvenience, it's an OS limit. 

I'm working on a new  feature (to be able to run the same executable parallel) and this change is kind of a requirement. However I have some idea how to bypass this limitation. 

Check 2.9.2 version in the next 24 hours. 
Would you verify that it solves it please?

(It was funny to see you twitch about my extension. ^^)

**Comment by rhymu8354** [comment]
Thanks for the quick reply!  I will definitely try out 2.9.2 to see if it works for me.

Yes, I use your extension pretty much every stream.  I recommend it to any viewers who ask about it.  It's been extremely valuable in quickly finding, running, and debugging tests.

**Comment by rhymu8354** [comment]
I just tried it out on 2.9.4 and it seems to work again!  Thanks so much!!

**Comment by matepek** [comment]
Great.

In 2.9.5 GoogleTest parsing will work on windows too. I wasn't aware of the [situation](https://github.com/google/googletest/blob/master/googletest/src/gtest.cc#L3121) at all. 

I was thinking a bit about your twitch. I think I implemented several good features. If you feel you are missing something just let me know. It might exists or can be implemented.

**Comment by rhymu8354** [comment]
While testing today I noticed something I didn't see before, and was pleasantly surprised: the lines in my tests where test asserts/expects are failing are now marked in red.  I hadn't noticed this before, so I'm thinking it's new?  I love it!

**Comment by matepek** [comment]
Cool. Actually I have this feature for ages but I haven't noticed that the gtest output format is different on win and unix-like so I wasn't aware that it only works for me until I watched your twitch and I was surprised of the lack of that feature. :) 
Now I generalized the processor so it is working on windows too. ^^

Let me draw your attention to some other features too maybe you found it interesting.
- on the upper right corner under the "..." menu there is an "Enable autorun". If it is turned on and you rebuild/touch your executable it will automatically rerun all the tests of it.
- if tests are depending on some resource file one can use `dependsOn`.
- if tests in the same executable don't use some resources exclusively one can boost the run time by allowing running the same executable with different test-subsets with `parallelizationLimit`
- with `testGrouping` one can group tests by regex even if those tests are in a different executable.
- oh and at last: check [this extension](https://marketplace.visualstudio.com/items?itemName=connorshea.vscode-test-explorer-status-bar). It is quite nice and works with my extension.

See [config](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/documents/configuration/executables.config.md) for details and ask me if you have any questions.
